core/src/main/scala/org/apache/spark/scheduler/Pool.scala

```diff
@@ -98,13 +97,12 @@ private[spark] class Pool(
   }
 
   override def getSortedTaskSetQueue: ArrayBuffer[TaskSetManager] = {
-    var sortedTaskSetQueue = new ArrayBuffer[TaskSetManager]
-    val sortedSchedulableQueue =
-      schedulableQueue.asScala.toSeq.sortWith(taskSetSchedulingAlgorithm.comparator)
-    for (schedulable <- sortedSchedulableQueue) {
-      sortedTaskSetQueue ++= schedulable.getSortedTaskSetQueue
-    }
-    sortedTaskSetQueue
+    schedulableQueue
```

Comment:
It compiles because of the cast, which might fail at runtime. Then maybe the tests didn't run? or didn't cover this code path actually